hail/python/hailtop/aiotools/fs/copier.py

```diff
@@ -394,19 +397,23 @@ async def copy_source(srcentry: FileListEntry) -> None:
 
             await self._copy_file_multi_part(sema, source_report, srcfile, await srcentry.status(), url_join(full_dest, relsrcfile), return_exceptions)
 
-        async def create_copies() -> List[Callable[[], Awaitable[None]]]:
+        async def create_copies() -> Tuple[List[Callable[[], Awaitable[None]]], int]:
             nonlocal srcentries
+            bytes_to_copy = 0
             if srcentries is None:
                 srcentries = await files_iterator()
             try:
-                return [
-                    functools.partial(copy_source, srcentry)
-                    async for srcentry in srcentries]
+                copy_thunks = []
+                async for srcentry in srcentries:
+                    bytes_to_copy += await (await srcentry.status()).size()  # this is almost never a syscall/net-request (afaik: only local symlinks)
```

Comment:
I don't understand the end of this comment. What do you mean this does a symlink?

Reply:
Getting the status and the size of an entry of a directory listing, AFAICT, only requires a system call or network request in one special case:

1. You have a LocalFS
2. The file in question is a symlink.

Linux doesn't eagerly retrieve the size of the target of a symlink (I think because it could be an arbitrary number of symlinks away).

Reply:
That's why status and size are `async`. If it wasn't for the local fs, they could be sync.

Maybe a better comment is:
> In cloud FSes, status and size never make a network request. In local FS, they can make system calls on symlinks. This line will be fairly expensive if copying a tree with a lot of symlinks.

Reply:
That's a better comment.

Reply:
Updated, thanks!